spec/controllers/admins_controller_spec.rb

```diff
@@ -0,0 +1,104 @@
+describe AdminsController do
+  describe '#assign' do
+    def go!
+      post 'assign'
+    end
+
+    context 'logged out' do
+      it 'redirects' do
+        go!
+        expect(response).to redirect_to('/')
+      end
+    end
+
+    context 'logged in' do
+      let!(:admin) { FactoryGirl.create(:admin, :activated) }
+      let!(:grant) { FactoryGirl.create(:grant) }
+      let!(:grant_2) { FactoryGirl.create(:grant) }
+
+      before do
+        FactoryGirl.create_list(:grant_submission, 16, funding_decision: false, grant: grant)
+        FactoryGirl.create_list(:grant_submission, 10, funding_decision: false, grant: grant_2)
+
+        FactoryGirl.create(:voter, :activated)
+        FactoryGirl.create_list(:voter, 4, :activated, verified: false)
+        FactoryGirl.create_list(:voter, 10, :activated)
+
+        # assign to first grant
+        15.times do |n|
+          FactoryGirl.create(:grants_voter, voter: Voter.find(n + 1), grant: grant)
+        end
+
+        # assign to second grant
+        2.times do |n|
+          FactoryGirl.create(:grants_voter, voter: Voter.find(n + 6), grant: grant_2)
+        end
+      end
+
+      before { sign_in_admin(admin.id) }
+
+      # copied as-is from existing tests, needs to be reworked
+      it 'matches expectations' do
```

Comment:
This test was copied over from the existing code and 'rspecized'. This part of the codebase should probably be moved out to a well tested class in `lib`, it is a lot of logic to have in a controller.